speaker/main.go

```diff
@@ -209,6 +211,16 @@ func (c *controller) SetBalancer(l log.Logger, name string, svc *v1.Service, eps
 		l.Log("bug", "true", "msg", "internal error: allocated IP has no matching address pool")
 		return c.deleteBalancer(l, name, "internalError")
 	}
+	validForNode := false
+	for _,ns := range pool.NodeSelectors {
+		if ns.Matches(c.nodeLabels) {
+			validForNode = true
+		}
+	}
+	if !validForNode {
+		l.Log("op", "setBalancer", "error", "assigned IP not allowed in this node by config")
+		return c.deleteBalancer(l, name, "ipNotAllowed")
+	}
```

Comment:
I see how this will do the desired Node filtering, but I think it will leave some Services orphaned with no Node at all, at least in layer 2 mode.

The algorithm used in the layer 2 controller will need to be updated to account for this.  I think it could decide that the Node that should be the leader is one not allowed by the node selector policy.  It will also need to account for the possibility that none of the endpoints are active on Nodes that are allowed for a given address pool.